packages/alchemy/src/react/components/auth/sections/InjectedProvidersAuth.tsx

```diff
@@ -0,0 +1,56 @@
+import type { Connector } from "@wagmi/core";
+import { useChain } from "../../../hooks/useChain.js";
+import { useConnect } from "../../../hooks/useConnect.js";
+import { Button } from "../../button.js";
+import { useAuthContext } from "../context.js";
+
+export const InjectedProvidersAuth = () => {
+  const { chain } = useChain();
+  const { connectors, connect } = useConnect({
+    onMutate: ({ connector }) => {
+      setAuthStep({ type: "eoa_connect", connector: connector as Connector });
```

Comment:
because of the way we use this below, this type cast is fine. I'm gonna add a comment about this in the code